### PR TITLE
Composer support

### PR DESCRIPTION
**Pull request by tomjn**
Adds a `composer.json` file, and updates the URL to use HTTPS in the plugin description

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
-.DS_Store
\ No newline at end of file
+.DS_Store
+/vendor/
```
Comment (rmccue) [inline .gitignore]: Also `composer.lock` maybe?

File: social-counts.php [hunk 2]
```diff
@@ -5,7 +5,7 @@
 Description: Adds the # of times a post has been shared on major social networks as post meta.
 Version: 2.0
 Author: Human Made
-Author URI: http://hmn.md/
+Author URI: https://hmn.md/
```
Comment (rmccue) [inline social-counts.php]: Could use `humanmade.com` instead

**Review by rmccue — APPROVED** [review]
lgtm

**Comment by svandragt** [comment]
If you don’t mind me asking after the fact, what’s the rationale of adding composer.lock to the gitignore? People would like to fallback to a tested set of versions they can install right in case a composer update breaks something?

**Comment by tomjn** [comment]
I can change that, but given that there are no real dependencies of the plugin itself other than composer installers and what is in `master` on the coding standards repo, the contents of `composer.lock` are somewhat irrelevant if we're checking if things work or not.

For a production install, only this repo itself is loaded and run

**Comment by rmccue** [comment]
`composer.lock` only operates on the root project, so for a library (or library-like thing such as a plugin), it's useless.

**Pull request merged**